crates/sui-types/src/transaction.rs

```diff
@@ -947,6 +948,32 @@ impl ProgrammableTransaction {
             command.validity_check(config)?;
         }
 
+        // A command that uses Random can only be followed by TransferObjects or MergeCoins.
+        if config.enable_randomness_ptb_limits() {
+            // Check if there is a random object in the input objects
+            if let Some(random_index) = inputs.iter().position(|obj| {
+                matches!(obj, CallArg::Object(ObjectArg::SharedObject { id, .. }) if *id == SUI_RANDOMNESS_STATE_OBJECT_ID)
+            }) {
+                let mut used_random_object = false;
+                let random_index = random_index as u16;
+                for command in commands {
+                    if !used_random_object {
+                        let has_random = if let Command::MoveCall(c) = command {
```

Comment:
Is it possible for someone to first call MakeMoveVec to put a randomness object in a vector and then use it?
It might be clearer to add an API like `is_arg_used(arg)` to `Command`.

Reply:
Changed to include here all commands with Random though in reality those other cases are not problematic - we want to make sure that no other Move calls can be done after a call to an "honest" function that accept &Random as an input, to make sure its outputs or side effects cannot be used in following commands.